src/main/java/org/opensearch/neuralsearch/util/prune/PruneUtils.java

```diff
@@ -0,0 +1,289 @@
+    public static Map<String, Float> pruneSparseVector(PruneType pruneType, float pruneRatio, Map<String, Float> sparseVector) {
+        if (Objects.isNull(pruneType) || Objects.isNull(pruneRatio)) {
+            throw new IllegalArgumentException("Prune type and prune ratio must be provided");
+        }
+
+        if (Objects.isNull(sparseVector)) {
+            throw new IllegalArgumentException("Sparse vector must be provided");
+        }
+
+        for (Map.Entry<String, Float> entry : sparseVector.entrySet()) {
+            if (entry.getValue() <= 0) {
+                throw new IllegalArgumentException("Pruned values must be positive");
+            }
+        }
+
+        switch (pruneType) {
+            case TOP_K:
+                return pruneByTopK(sparseVector, (int) pruneRatio, false).v1();
+            case ALPHA_MASS:
+                return pruneByAlphaMass(sparseVector, pruneRatio, false).v1();
+            case MAX_RATIO:
+                return pruneByMaxRatio(sparseVector, pruneRatio, false).v1();
+            case ABS_VALUE:
+                return pruneByValue(sparseVector, pruneRatio, false).v1();
+            default:
+                return sparseVector;
+        }
+    }
+
+    /**
+     * Validates whether a prune ratio is valid for a given prune type.
+     *
+     * @param pruneType The type of prune strategy
+     * @param pruneRatio The ratio or threshold to validate
+     * @return true if the ratio is valid for the given prune type, false otherwise
+     * @throws IllegalArgumentException if prune type is null
+     */
+    public static boolean isValidPruneRatio(PruneType pruneType, float pruneRatio) {
+        if (pruneType == null) {
+            throw new IllegalArgumentException("Prune type cannot be null");
+        }
+
+        switch (pruneType) {
+            case TOP_K:
+                return pruneRatio > 0 && pruneRatio == Math.floor(pruneRatio);
+            case ALPHA_MASS:
+            case MAX_RATIO:
+                return pruneRatio >= 0 && pruneRatio < 1;
+            case ABS_VALUE:
+                return pruneRatio >= 0;
+            default:
+                return true;
+        }
+    }
+
+    /**
+     * Get description of valid prune ratio for a given prune type.
+     *
+     * @param pruneType The type of prune strategy
+     * @throws IllegalArgumentException if prune type is null
+     */
+    public static String getValidPruneRatioDescription(PruneType pruneType) {
+        if (pruneType == null) {
+            throw new IllegalArgumentException("Prune type cannot be null");
+        }
+
+        switch (pruneType) {
+            case TOP_K:
+                return "prune_ratio should be positive integer.";
+            case MAX_RATIO:
+            case ALPHA_MASS:
+                return "prune_ratio should be in the range [0, 1).";
+            case ABS_VALUE:
+                return "prune_ratio should be non-negative.";
+            default:
+                return "";
```

Comment:
I think it's feasible to add description for case of NONE.

Reply:
ack